data_structures/priority_queue.hpp

```diff
@@ -0,0 +1,195 @@
+/**
+ * @file
+ * @author [Tuan Ho](https://github.com/ttuanho)
+ * @brief Implementation of [Priority
+ * queue](https://en.wikipedia.org/wiki/Priority_queue) an abstract data type
+ * similar to regular queue or stack data structure in which each element
+ * additionally has a "priority" associated with it.
+ * @note improvised from queue.h
+ * @see queue.h
+ */
+
+#ifndef DATA_STRUCTURES_PRIORITY_QUEUE_H_
+#define DATA_STRUCTURES_PRIORITY_QUEUE_H_
```

Comment:
I think these must be `HPP`.
```suggestion
#ifndef DATA_STRUCTURES_PRIORITY_QUEUE_HPP_
#define DATA_STRUCTURES_PRIORITY_QUEUE_HPP_
```